frontend/packages/ux-editor/src/components/config/editModal/EditImage/EditImage.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Tabs } from '@digdir/designsystemet-react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { ComponentType } from 'app-shared/types/ComponentType';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useGetAllImageFileNamesQuery } from 'app-shared/hooks/queries/useGetAllImageFileNamesQuery';
+import { useTranslation } from 'react-i18next';
+import classes from './EditImage.module.css';
+import { useDeleteImageMutation } from 'app-shared/hooks/mutations/useDeleteImageMutation';
+import { LocalImage } from './LocalImage';
+import { ExternalImage } from './ExternalImage';
+
+enum ImageTab {
+  Import = 'import',
+  ExternalUrl = 'externalUrl',
+}
+
+export interface EditImageProps extends IGenericEditComponent<ComponentType.Image> {}
+
+export const EditImage = ({ component, handleComponentChange }: EditImageProps) => {
+  const { t } = useTranslation();
+  const [tab, setTab] = useState<string>(ImageTab.Import);
+  const { org, app } = useStudioEnvironmentParams();
+  const {
+    data: imageFileNames,
+    isPending: imageFileNamesArePending,
+    refetch: refetchImageFileNames,
+  } = useGetAllImageFileNamesQuery(org, app);
+  const { mutate: deleteImageFromLibrary } = useDeleteImageMutation(org, app);
+
+  const fileName = extractFileNameFromImageSrc(component.image?.src?.nb, org, app);
+  const imageOriginsFromLibrary = !imageFileNamesArePending && imageFileNames.includes(fileName);
+
+  const handleImageChange = async (imageSource: string, fromUrl: boolean = false) => {
```

Comment:
Lag gjerne to separate funksjoner i stedet for å sende inn en `boolean` som et parameter.

Reply:
Det kan jeg godt, men av nysgjerrighet - hvorfor er det et bedre pattern? Funkjsonen som skal utføres er jo lik i begge caser, men formatet på bildestien varierer litt i hvert tilfelle bare 🤔 

Jeg prøvde meg uansett på å løse det på en annen måte. Jeg legger til 'wwwroot/' fra ImportImage istedenfor så den allerede er en del av imageSource når det kommer tilbake til dette nivået 👍

Reply:
Å sende inn et boolsk parameter kan ofte se ut som en enklere løsning, ja. Men det går på bekostning av lesbarheten. Det er ikke mulig å forstå hva `handleImageChange('abc.png', true)` gjør uten å se på definisjonen av funksjonen. `handleImageChangeFromUrl('abc.png')` er derimot mye mer lesbart. I tillegg blir selve funksjonen enklere siden den får et mer spesifikt ansvarsområde.

Reply:
Internett er for øvrig fullt av artikler med gode forklaringer på dette. Her er et eksempel jeg kom over med et raskt Google-søk: https://alexkondov.com/should-you-pass-boolean-to-functions/